File: benchmark/corpus/javascript/xss_vuln_1.js
// VULN: CWE-79 — DOM-based XSS via innerHTML with URL parameter
function showGreeting() {
    const urlParams = new URLSearchParams(window.location.search);
    const name = urlParams.get('name') || 'World';
    document.getElementById('greeting').innerHTML = 'Hello, ' + name;
}

window.onload = showGreeting;

File: benchmark/corpus/python/xss_vuln_1.py
"""VULN: CWE-79 — XSS / SSTI via render_template_string with user input."""
import flask

app = flask.Flask(__name__)


@app.route("/greet")
def greet():
    name = flask.request.args.get("name", "World")
    template = f"<h1>Hello, {name}!</h1>"
    return flask.render_template_string(template)
